narwhal/types/src/primary.rs

```diff
@@ -780,6 +840,8 @@ impl fmt::Display for BlockErrorKind {
 pub struct WorkerOurBatchMessage {
     pub digest: BatchDigest,
     pub worker_id: WorkerId,
+    /// The time when the batch was created
```

Comment:
Can this just be the actual metadata struct instead of the one timestamp field? Or is there some reason we wouldn't want the primary to have access to other metadata, too, if we add it? I am worried that if you extract a single field instead of putting `Metadata` in here then it's not as obvious that all of your stern warnings about not relying on anything in metadata as "trusted" may not be noticed

Reply:
I think we can use the `Metadata` here as well - thanks @aschran I'll do so